### DISCUSSION TITLE
Conditional Metrics

### DISCUSSION THREAD

**Discussion by murrant** [prometheus/snmp_exporter#854] [Q&A] [answered]
Is there any way to do conditional metrics?

For example, if ifAdminStatus or ifOperStatus are "down" or "lowerLayerDown", don't collect ifInOctets, ifOutOctets, etc?

This can be tricky, because you would snmpwalk ifAdminStatus and ifOperStatus and snmpget of other OIDs if any ports are excluded.

This can be important for devices with many ports or devices with limited processing power.

**Comment by SuperQ** [comment, marked as answer]
No, this is not really an option. It also is problematic because it would cause metrics to appear and disappear in the case of a flapping port. This would cause problems for Prometheus (Staleness handling) and would be weird for queries.

You're better off splitting / filtering your walks. For example on JunOS it's not a CPU problem, but a data gathering speed problem. For these I wrote a more limited set of IF-MIB walks that reduce the load on the device and spread the gathering over time.

**Reply by murrant** [reply]
I found your junos yaml file and we were already only fetching limited columns of the ifTable+ifXTable and it wasn't enough.  Sounds like there are no more options to trim down the amount of data collected more than that.
